shotover-proxy/tests/runner/observability_int_tests.rs

```diff
@@ -26,16 +90,34 @@ async fn test_metrics() {
         .await
         .unwrap();
 
-    let client = hyper::Client::new();
-    let uri = "http://localhost:9001/metrics".parse().unwrap();
-    let res = client.get(uri).await.unwrap();
-    let body_bytes = hyper::body::to_bytes(res.into_body()).await.unwrap();
-    let body = String::from_utf8(body_bytes.to_vec()).unwrap();
-
-    // If the body contains these substrings, we can assume metrics are working
-    assert!(body.contains("# TYPE shotover_transform_total counter"));
-    assert!(body.contains("# TYPE shotover_chain_total counter"));
-    assert!(body.contains("# TYPE shotover_available_connections gauge"));
-    assert!(body.contains("# TYPE shotover_transform_latency summary"));
-    assert!(body.contains("# TYPE shotover_chain_latency summary"));
+    redis::cmd("GET")
+        .arg("the_key")
+        .query_async::<_, ()>(&mut connection)
+        .await
+        .unwrap();
+
+    std::thread::sleep(time::Duration::from_secs(1));
```

Comment:
Oh did we end up actually needing this sleep?

Reply:
I guess we could loop now that we are counting the lines?

Reply:
nah, sleep is fine for now if we need it.
I just thought the issue turned out to not be a race condition.